packages/backend-glr/src/Happy/Backend/GLR/ProduceCode.lhs

```diff
@@ -202,7 +202,7 @@ Extract the string that comes before the module declaration...
 >     . table_text
 
 >   lib_content lib_text
->    = let (pre,_drop_me : post) = break (== "fakeimport DATA") $ lines lib_text
+>    = let (pre,post) = break (== "fakeimport DATA") $ lines lib_text
```

Comment:
Should we make such a change?  I think it is hurting code comprehensibility more than helping it.
The current formulation makes clear that the `_drop_me :  post` match MUST succeed, even if GHC does not see it.
The new formulation clouds this insight.
So, I am against such a change.

Reply:
Yes it would be better to add an explicit pattern match with an `error ...`. It would be even better to try to use `NonEmpty`.

Reply:
`NonEmpty` is available from `base-compat` for old GHC versions back to 7.0 (`base-4.3`), so that would not be a problem for keeping all currently supported GHC versions.